### PR TITLE
LMBQ-329: Adding Slick carousel

### PR DESCRIPTION
**Pull request by barthamark**
[LMBQ-329](https://lombiq.atlassian.net/browse/LMBQ-329)

[LMBQ-329]: https://lombiq.atlassian.net/browse/LMBQ-329?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: Lombiq.UIKit/Views/SlickCarousel.cshtml [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+</div>
+
+<script at="Foot">
+    jQuery(function ($) {
+        const $carouselContainer = $("#@carouselId");
+        const $carousel = $carouselContainer.find(".slickCarousel__carousel");
+        const $carouselArrows = $carouselContainer.find(".slickCarousel__arrows");
+
+        $carousel.on("init", () => {
+            // Carousel initializes elements with empty id attributes, which triggers HTML validation errors.
```
Comment (DemeSzabolcs) [inline Lombiq.UIKit/Views/SlickCarousel.cshtml:43]: Is this always the case? Can't you add ID-s to the elements, or preventing adding the id-s?

If it's always added and there is no way to fill them up (so they will be always empty), then open an issue about this here: https://github.com/kenwheeler/slick

File: Lombiq.UIKit.Tests.UI/Extensions/TestCaseUITestContextExtensions.cs [hunk 2]
```diff
@@ -33,4 +34,7 @@ public static async Task TestDropdownButtonAsync(this UITestContext context)
         await context.ClickReliablyOnAsync(By.CssSelector("#buttonContainerId2 button"));
         context.Get(By.CssSelector("#buttonContainerId2 > ul")).Displayed.ShouldBeTrue();
     }
+
+    public static async Task TestSlickCarouselAsync(this UITestContext context) =>
```
Comment (DemeSzabolcs) [inline Lombiq.UIKit.Tests.UI/Extensions/TestCaseUITestContextExtensions.cs]: https://github.com/Lombiq/Open-Source-Orchard-Core-Extensions/actions/runs/8883842545/job/24391572728?pr=749#step:6:162

**Review by DemeSzabolcs — CHANGES_REQUESTED** [review]

File: Lombiq.UIKit/Views/SlickCarousel.cshtml (hunk 1 above)
Comment (DemeSzabolcs) [reply Lombiq.UIKit/Views/SlickCarousel.cshtml:43]: @barthamark

Comment (barthamark) [reply Lombiq.UIKit/Views/SlickCarousel.cshtml:43]: Done: https://github.com/kenwheeler/slick/issues/4296 I've also added it to the code as a comment.

File: Lombiq.UIKit/Views/SlickCarousel.cshtml [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
```
Comment (DemeSzabolcs) [inline Lombiq.UIKit/Views/SlickCarousel.cshtml:43]: Link the opened issue in the comment.

**Review by DemeSzabolcs — CHANGES_REQUESTED** [review]

**Review by DemeSzabolcs — APPROVED** [review]

**Review by DemeSzabolcs — APPROVED** [review]

**Pull request merged**